store/tikv/coprocessor.go

```diff
@@ -1176,3 +1168,50 @@ func (it copErrorResponse) Next(ctx context.Context) (kv.ResultSubset, error) {
 func (it copErrorResponse) Close() error {
 	return nil
 }
+
+// copResponseCollector is used to collect the resp from task respCh into its own resp
+type copResponseCollector struct {
+	respChan chan *copResponse
+	tasks    []*copTask
+	sendRate *rateLimit
+	finishCh <-chan struct{}
+}
+
+func (c *copResponseCollector) run() {
```

Comment:
it should use with.recovery to record the heap if the goroutine exit by accident.